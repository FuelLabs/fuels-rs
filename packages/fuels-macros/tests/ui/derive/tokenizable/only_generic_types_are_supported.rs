use fuels_macros::Tokenizable;

#[derive(Tokenizable)]
enum SomeEnum<const T: usize> {
    A,
}

#[derive(Tokenizable)]
enum AnotherEnum<'a> {
    A(&'a u64),
}

#[derive(Tokenizable)]
struct SomeStruct<const T: usize> {}

#[derive(Tokenizable)]
struct AnotherStruct<'a> {
    a: &'a u64,
}

fn main() {}
