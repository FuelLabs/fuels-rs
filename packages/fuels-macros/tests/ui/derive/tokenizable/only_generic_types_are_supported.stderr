error: Const generics not supported
 --> tests/ui/derive/tokenizable/only_generic_types_are_supported.rs:4:15
  |
4 | enum SomeEnum<const T: usize> {
  |               ^^^^^^^^^^^^^^

error: Lifetimes not supported
 --> tests/ui/derive/tokenizable/only_generic_types_are_supported.rs:9:18
  |
9 | enum AnotherEnum<'a> {
  |                  ^^

error: Const generics not supported
  --> tests/ui/derive/tokenizable/only_generic_types_are_supported.rs:14:19
   |
14 | struct SomeStruct<const T: usize> {}
   |                   ^^^^^^^^^^^^^^

error: Lifetimes not supported
  --> tests/ui/derive/tokenizable/only_generic_types_are_supported.rs:17:22
   |
17 | struct AnotherStruct<'a> {
   |                      ^^
